### PR TITLE
feat: bring additional alert_context to AWS rules which had none

### PR DESCRIPTION
**Pull request by edyesed**
### Background
This change adds an alert_context to all AWS rules which did not have `alert_context()` already defined. This will ensure that the alert_context field on alerts holds the eventName, eventSource, userIdentity, sourceIps, useragent info to surfaced alerts.

This change enriches the data that flows into downstream systems which receive alert data.

### Changes

* add global `panther_aws_helpers.aws_rule_context()`
* Use new rule context helper in aws rules

### Testing

* make test

### CODE REVIEW

**Comment by edyesed** [comment]
references this asana task https://app.asana.com/0/1201295359000496/1202764735384190/f

File: aws_guardduty_rules/aws_guardduty_high_sev_findings.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
+from panther_aws_helpers import aws_rule_context
+
 def rule(event):
     return 7.0 <= float(event.get("severity", 0)) <= 8.9
 
-
 def dedup(event):
     return event.get("id")
 
-
```
Comment (wey-chiang) [inline aws_guardduty_rules/aws_guardduty_high_sev_findings.py:8]: [PEP8](https://peps.python.org/pep-0008/#blank-lines) disagrees with this -- there should be two lines surrounding top-level funcs and class definitions

File: aws_s3_rules/aws_s3_unauthenticated_access.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
 # A list of buckets where authenticated access is expected
-AUTH_BUCKETS = {"example-bucket"}
+from panther_aws_helpers import aws_rule_context
 
+AUTH_BUCKETS = {"example-bucket"}
```
Comment (wey-chiang) [inline aws_s3_rules/aws_s3_unauthenticated_access.py:4]: We shouldn't have comments at the very top of the file unless it's for a license

The line
`from panther_aws_helpers import aws_rule_context`
Should be on line 1 and the comment current there should be above `AUTH_BUCKETS`

**Review by wey-chiang — COMMENTED** [review]
@edyesed could you move the `aws_rule_context` to be in the `panther_base_helpers` global helpers file? 

That file holds all of the alert context functions we have (Okta, CrowdStrike, and Slack) defined and it should be grouped as such. This would also reduce tech debt as we would need to package the new helper in the packs and creating a new global just to house a single function doesn't seem right.

File: global_helpers/panther_base_helpers.py [hunk 3]
```diff
@@ -248,6 +248,16 @@ def aws_strip_role_session_id(user_identity_arn):
     return user_identity_arn
 
 
+def aws_rule_context(event: dict):
+    return {
+        "eventName": event.get("eventName", ""),
+        "recipientAccountId": event.get("recipientAccountId", ""),
+        "sourceIPAddress": event.get("sourceIPAddress", ""),
+        "userAgent": event.get("userAgent", ""),
+        "userIdentity": event.get("userIdentity", ""),
```
Comment (wey-chiang) [inline global_helpers/panther_base_helpers.py:257]: nit: add default values
```suggestion
        "eventName": event.get("eventName", "<MISSING_EVENT_NAME>"),
        "recipientAccountId": event.get("recipientAccountId", "<MISSING_ACCOUNT_ID>"),
        "sourceIPAddress": event.get("sourceIPAddress", "<MISSING_SOURCE_IP>"),
        "userAgent": event.get("userAgent", "<MISSING_USER_AGENT>"),
        "userIdentity": event.get("userIdentity", "<MISSING_USER_IDENTITY>"),
```

**Review by wey-chiang — APPROVED** [review]
One nit with suggested changes but overall LGTM

**Review by wey-chiang — APPROVED** [review]

**Pull request merged**